pinot-core/src/main/java/org/apache/pinot/core/transport/ServerRoutingInstance.java

```diff
@@ -103,16 +93,16 @@ public boolean equals(Object o) {
       return false;
     }
     ServerRoutingInstance that = (ServerRoutingInstance) o;
-    // NOTE: Only check hostname, port and tableType for performance concern because they can identify a routing
+    // NOTE: Only check hostname and port for performance concern because they can identify a routing
     //       instance within the same query
-    return _hostname.equals(that._hostname) && _port == that._port && _tableType == that._tableType;
+    return _hostname.equals(that._hostname) && _port == that._port;
   }
 
   @Override
   public int hashCode() {
-    // NOTE: Only check hostname, port and tableType for performance concern because they can identify a routing
+    // NOTE: Only check hostname and port for performance concern because they can identify a routing
     //       instance within the same query
-    return 31 * 31 * _hostname.hashCode() + 31 * Integer.hashCode(_port) + _tableType.hashCode();
+    return 31 * _hostname.hashCode() + Integer.hashCode(_port);
```

Comment:
This will make the hashcode inconsistent between two different versions of broker and servers. Can you double check if this has any side effects?

Reply:
This will have one major intentional implication, which is that we would no longer open 2 channels from each broker to each server (1 for REALTIME, 1 for OFFLINE). In turn, that would cause the peak network traffic on a channel to at worst double, assuming all queries were to hybrid tables.  I'm not certain whether that presents an issue.

Reply:
Probably worth stating that the motivation in doing so is to be able to have a single broker request fanout to an arbitrary number of requests to any single server involved in query processing.  This unblocks a major hurdle in the ability to support Logical Tables[1].  

[1] https://github.com/apache/pinot/issues/10712